### ISSUE TITLE
same build error Win10

### ISSUE THREAD

**Issue by clive-g-brown** [jramapuram/hal#13]
Only change I made was to Csrgo.toml to update spmc as the version in there is gone. ArrayFire-rust alone compiles fine

error[E0277]: `(dyn std::ops::Fn(defines::AfError) + 'static)` cannot be shared between threads safely
  --> arrayfire-rust\src\error.rs:25:1
   |
25 | pub static DEFAULT_HANDLE_ERROR: &'static ErrorCallback = &handle_error_general;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `(dyn std::ops::Fn(defines::AfError) + 'static)` cannot be shared between threads safely
   |
   = help: within `&'static (dyn std::ops::Fn(defines::AfError) + 'static)`, the trait `std::marker::Sync` is not implemented for `(dyn std::ops::Fn(defines::AfError) + 'static)`
   = note: required because it appears within the type `&'static (dyn std::ops::Fn(defines::AfError) + 'static)`
   = note: shared static variables must have a type that implements `Sync`

**Comment by jramapuram** [comment]
This is most likely because you are probably using a newer Rust version (untested). 
Can you try using rustup to get 1.8 and try it?

**Comment by clive-g-brown** [comment]
thanks, the highest version I can find is 1.44.0

**Comment by jramapuram** [comment]
Yup, its a little counter-intuitive, but 1.8.0 is older than 1.44.0
Try:

```bash
rustup toolchain install 1.8.0
```

**Comment by clive-g-brown** [comment]
yeah tried that, doesn't seem to find it. never mind.

**Comment by jramapuram** [comment]
Strange, seems to work for me: 

![image](https://user-images.githubusercontent.com/8204807/82336723-35f78680-99eb-11ea-9a60-dcfd49fdcf55.png)

In either case, I have not worked on this repo in a long time (since tensorflow / pytorch) as I'm working on my PhD. If you did want to work on it I would suspect it wouldn't work out-of-the-box anyhow.
